docker/ci/build.ps1

```diff
@@ -0,0 +1,106 @@
+$ErrorActionPreference = 'Stop'
+
+function Get-CurrentDirectory
+{
+  $thisName = $MyInvocation.MyCommand.Name
+  [IO.Path]::GetDirectoryName((Get-Content function:$thisName).File)
+}
+
+function Get-ContainerVersion
+{
+  # shallow repositories need to pull remaining code to `git describe` correctly
+  if (Test-Path "$(git rev-parse --git-dir)/shallow")
+  {
+    git fetch --unshallow
+  }
+
+  # tags required for versioning
+  git fetch origin 'refs/tags/*:refs/tags/*'
+  (git describe) -replace '-.*', ''
+}
+
+# installs gems for build and test and grabs base images
+function Invoke-ContainerBuildSetup
+{
+  Push-Location (Get-CurrentDirectory)
+  docker pull ubuntu:16.04
+  Pop-Location
+}
+
+function Lint-Dockerfile($Path)
+{
+  hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001 $Path
+}
+
+function Build-Container(
+  $Namespace = 'puppet',
+  $Version = (Get-ContainerVersion),
+  $Vcs_ref = $(git rev-parse HEAD),
+  $Base = 'ubuntu')
+{
+  Push-Location (Join-Path (Get-CurrentDirectory) '..')
+
+  $build_date = (Get-Date).ToUniversalTime().ToString('o')
+  $subdir = if ($Base -eq 'alpine') { 'windows-build' } else { '' }
+  $docker_args = @(
+    '--pull',
+    '--build-arg', "vcs_ref=$Vcs_ref",
+    '--build-arg', "build_date=$build_date",
+    '--build-arg', "version=$Version",
+    '--build-arg', "namespace=$Namespace",
+    '--file', "puppet-agent-$Base/$subdir/Dockerfile",
+    '--tag', "$Namespace/puppet-agent-${Base}:$Version",
+    '--tag', "$Namespace/puppet-agent-${Base}:latest"
```

Comment:
If we're just using these for local testing we probably don't need the latest tag